demo_nodes_cpp/src/topics/talker.cpp

```diff
@@ -50,7 +50,9 @@ class Talker : public rclcpp::Node
         pub_->publish(std::move(msg_));
       };
     // Create a publisher with a custom Quality of Service profile.
-    rclcpp::QoS qos(rclcpp::KeepLast(7));
+    // Extra pair of parentheses to address vexing parse problem.
+    // Or use uniform initialization.
```

Comment:
```suggestion
    // Uniform initialization is suggested so it can be trivially changed to rclcpp::KeepAll{} if the user wishes.
    // (rclcpp::KeepLast(7) -> rclcpp::KeepAll() fails to compile)
```

Reply:
> why rclcpp::QoS qos(rclcpp::KeepAll()); fails to compile.

i was there too, as far as i know for now, it is about C++ compiler parse. (i am not sure if this is a problem, but it really bothers me and took time to check.) So i just thought it would be better to file this to avoid possible future problem.